lms/djangoapps/completion_api/tests/test_views.py

```diff
@@ -204,3 +204,80 @@ def test_staff_access(self):
         self.client.force_authenticate(user)
         response = self.client.get('/api/completion/v0/course/?user={}'.format(self.test_user.username))
         self.assertEqual(response.status_code, 200)
+
+
+@override_settings(STUDENT_GRADEBOOK=True)
+class CompletionBlockUpdateView(SharedModuleStoreTestCase):
+    """
+    Test that the CompletionView renders completion data properly.
+
+    Ensure that it handles authorization as well.
+    """
+
+    usage_key = 'i4x://edX/toy/video/sample_video'
+    incorrect_usage_key = 'i4x://edX/toy/video/another'
+
+    @classmethod
+    def setUpClass(cls):
+        super(CompletionBlockUpdateView, cls).setUpClass()
+        cls.course = ToyCourseFactory.create()
+        cls.another_course = ToyCourseFactory.create()
+
+    def setUp(self):
+        super(CompletionBlockUpdateView, self).setUp()
+        self.test_user = UserFactory.create()
+        self.client = APIClient()
+        self.client.force_authenticate(user=self.test_user)
+        self.update_url = '/api/completion/v0/course/{}/blocks/{}'.format(self.course.id, self.usage_key)
+
+    def test_create_view(self):
+        completion_query_url = '/api/completion/v0/course/edX/toy/2012_Fall/?requested_fields=sequential'
+
+        response = self.client.get(completion_query_url)
+        self.assertEqual(response.status_code, 200)
+        expected = {
+            'course_key': 'edX/toy/2012_Fall',
+            'completion': {
+                'earned': 1.0,
+                'possible': 12.0,
+                'ratio': 1 / 12,
+            },
+            'sequential': {
+                'course_key': 'edX/toy/2012_Fall',
+                'block_key': u'i4x://edX/toy/sequential/vertical_sequential',
+                'completion': {
+                    'earned': 0.0,
+                    'possible': 12.0,
+                    'ratio': 0.0,
+                },
+            }
+        }
```

Comment:
In this test case, we're not really testing the format of the completion query API, so I don't think it's ideal to test the exact output of the API. Rather the before-update checks should match the after-update checks:

```python

before_response = self.client.get(completion_query_url)
self.assertEqual(before_response.status_code, 200)
self.assertEqual(before_response['completion']['earned'], 0)
self.assertEqual(before_response['sequential'][0]['earned'], 0)

response = self.client.post(self.update_url, {'completion': 1})
self.assertEqual(response.status_code, 201)

after_response = self.client.get(completion_query_url)
self.assertEqual(after_response.status_code, 200)
self.assertEqual(after_response['completion']['earned'], 1)
self.assertEqual(after_response['sequential'][0]['earned'], 1)
```